ui/components/Metadata.tsx

```diff
@@ -41,40 +40,62 @@ function Metadata({ metadata = [], labels = [], className }: Props) {
         </Link>
       );
   });
+  return metadataCopy;
+};
 
+const MetadataSection: React.FC<{ title: string; items: [string, any][] }> = ({
+  title,
+  items,
+}) => {
   return (
-    <Flex wide column className={className} gap="16">
-      {metadataCopy.length > 0 && (
-        <Flex column gap="8">
-          <Text size="large" color="neutral30">
-            Metadata
-          </Text>
-          <InfoList items={metadataCopy} />
-          <Spacer padding="small" />
+    <Flex column gap="8">
+      <Text size="large" color="neutral30">
+        {title}
+      </Text>
+      {title === "Labels" ? (
+        <Flex wide start wrap gap="4">
+          {items.map((label, index) => {
```

Comment:
Is it possible for `items` to be `null` or `undefined` here? If so, this will explode.

Reply:
The labels prop gets set to `[]` if undefined, and then it gets checked for length before rendering the section at all